pyanaconda/modules/storage/bootloader/systemd.py

```diff
@@ -0,0 +1,190 @@
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+#
+#
+#
+#
+# Systemd-boot is dead simple, it basically provides a second boot menu
+# and injects the kernel parms into an efi stubbed kernel, which in turn
+# loads it's initrd. As such there aren't any filesystem or drivers to worry
+# about as everything needed is provided by UEFI. Even the console remains
+# on the UEFI framebuffer, or serial as selected by UEFI. Basically
+# rather than trying to be another mini-os (like grub) and duplicate much of
+# what uefi provides, it simply utilizes the uefi services.
+#
+# Further, while we could keep stage1 (ESP) and stage2 (/boot) seperate it
+# simplifies things just to merge them and place the kernel/initrd on the
+# ESP. This requires a larger than normal ESP, but for now we assume that
+# this linux installer is creating the partitions, so where the space
+# is allocated doesn't matter.
+
+import os
+import re
+
+from pyanaconda.modules.storage.bootloader.base import BootLoader, BootLoaderError
+from pyanaconda.core import util
+from pyanaconda.core.configuration.anaconda import conf
+from pyanaconda.core.i18n import _
+from pyanaconda.core.path import open_with_perm
+from pyanaconda.product import productName
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+__all__ = ["SYSTEMD"]
+
+class SYSTEMD(BootLoader):
+    """Systemd-boot.
+
+    """
+    name = "SYSTEMD"
+    # oddly systemd-boot files are part of the systemd-udev package
+    # and in /usr/lib/systemd/boot/efi/systemd-boot[aa64].efi
+    # and the systemd stubs are in /usr/lib/systemd/linuxaa64.efi.stub
+    _config_file = "loader.conf"
+    _config_dir = "efi/loader/"
+    stage2_max_end = None
+
+    stage2_is_valid_stage1 = True
+    stage2_bootable = True
+    stage2_must_be_primary = False
+
+    # requirements for boot devices
+    stage2_device_types = ["partition"]
+
+    def __init__(self):
+        super().__init__()
+        self.encrypted_password = ""
+
+    #
+    # configuration
+    #
+
+    @property
+    def config_dir(self):
+        """ Full path to configuration directory. """
+        log.info("systemd.py: config_dir")
+        return "/boot/" + self._config_dir
```

Comment:
Maybe instead of hardcoding this patch you could use `bootctl --print-esp-path` and make `_config_dir = "/loader/"` instead?

That seems to be more aligned to what is expected, since AFAICT is mount point of the ESP + loader sub-dir.

Reply:
yes, I've fixed this (and its user) up to pick it up from bootctl, although I've not tried to see what happens if the esp is mounted in /efi since i think we should attempt to retain the RH /boot/efi mountpoint even if /boot ends up mostly empty (although symbol files and the like still tend to be there).

I have been testing it without a dedicated /boot partition now, with an additional tweak to the stage2 code.